lib/plug/ssl.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Plug.SSL do
+  @behaviour Plug
+
+  import Plug.Conn
+  alias Plug.Conn
+
+  def init(opts) do
+    hsts       = Keyword.get(opts, :hsts, true)
+    expires    = Keyword.get(opts, :expires, 31536000)
+    subdomains = Keyword.get(opts, :subdomains, false)
+    hsts = if hsts, do: [expires: expires, subdomains: subdomains], else: false
+
+    exclude = Keyword.get(opts, :exclude)
+    host = Keyword.get(opts, :host)
+
+    %{hsts: hsts,
+      exclude: exclude,
+      host: host}
+  end
+
+  def call(conn, config) do
+    cond do
+      config[:exclude] && config[:exclude].(conn) -> conn
```

Comment:
We can't support exclude because the way we configure most plugs (in the module body) does not allow anonymous functions to be passed. We can just remove this feature.